source/patterns/@aws-solutions-constructs/aws-apigateway-iot/README.md

```diff
@@ -0,0 +1,151 @@
+# aws-apigateway-iot module
+<!--BEGIN STABILITY BANNER-->
+
+---
+
+![Stability: Experimental](https://img.shields.io/badge/stability-Experimental-important.svg?style=for-the-badge)
+
+> All classes are under active development and subject to non-backward compatible changes or removal in any
+> future version. These are not subject to the [Semantic Versioning](https://semver.org/) model.
+> This means that while you may use them, you may need to update your source code when upgrading to a newer version of this package.
+
+---
+<!--END STABILITY BANNER-->
+
+| **Reference Documentation**:| <span style="font-weight: normal">https://docs.aws.amazon.com/solutions/latest/constructs/</span>|
+|:-------------|:-------------|
+<div style="height:8px"></div>
+
+| **Language**     | **Package**        |
+|:-------------|-----------------|
+|![Python Logo](https://docs.aws.amazon.com/cdk/api/latest/img/python32.png) Python|`aws_solutions_constructs.aws_apigateway_iot`|
+|![Typescript Logo](https://docs.aws.amazon.com/cdk/api/latest/img/typescript32.png) Typescript|`@aws-solutions-constructs/aws-apigateway-iot`|
+|![Java Logo](https://docs.aws.amazon.com/cdk/api/latest/img/java32.png) Java|`software.amazon.awsconstructs.services.apigatewayiot`|
+
+## Overview
+
+This AWS Solutions Construct implements an Amazon API Gateway REST API connected to AWS IoT pattern.
+
+This construct creates a scalable HTTPS proxy between API Gateway and AWS IoT. This comes in handy when wanting to allow legacy devices that do not support the MQTT or MQTT/Websocket protocol to interact with the AWS IoT platform.
+
+This implementation enables write-only messages to be published on given MQTT topics, and also supports shadow updates of HTTPS devices to allowed things in the device registry. It does not involve Lambda functions for proxying messages, and instead relies on direct API Gateway to AWS IoT integration which supports both JSON messages as well as binary messages.
+
+Here is a minimal deployable pattern definition:
+
+``` javascript
+const { ApiGatewayToIot } = require('@aws-solutions-constructs/aws-apigateway-iot');
+
+new ApiGatewayToIot(stack, 'ApiGatewayToIotPattern', {
+    apiGatewayToIotProps: {
+        iotEndpoint: 'a1234567890123-ats'
+    }
+});
+
+```
+
+## Initializer
+
+``` text
+new ApiGatewayToIot(scope: Construct, id: string, props: ApiGatewayToIotProps);
+```
+
+_Parameters_
+
+* scope [`Construct`](https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_core.Construct.html)
+* id `string`
+* props [`ApiGatewayToIotProps`](#pattern-construct-props)
+
+## Pattern Construct Props
+
+| **Name**     | **Type**        | **Description** |
+|:-------------|:----------------|-----------------|
+|iotEndpoint|`string`|The AWS IoT endpoint subdomain to integrate the API Gateway with (e.g a1234567890123-ats).|
```

Comment:
I used the naming inline with the IoT console terminology. A fully qualified endpoint would look like a1234567890123-ats.iot.us-east2-amazonaws.com, we just need the a1234567890123-ats part of the endpoint. I will add a check to pick only the required part (just in case user supplies fully qualified endpoint).